### PR TITLE
Refactored request unit tests to return promises

### PR DESCRIPTION
**Pull request by allouis**
[Master broke after the last test fixes](https://travis-ci.org/TryGhost/Ghost/jobs/445047196) - but with something new! here's a cleanup of the tests :relaxed: 

refs #7470

These tests had a random failure on travis for Node v6 w/ sqlite. I
figured I would update them to return promises, and remove state, to
make debugging easier, and hopefully fix any issues.

### CODE REVIEW

File: core/test/unit/lib/request_spec.js [hunk 1]
```diff
@@ -80,104 +67,103 @@ describe('Request', function () {
...
 
-        result = request(url, options)
-            .catch(function (err) {
-                requestMock.isDone().should.be.true();
-                should.exist(err);
-                err.statusMessage.should.be.equal('Not Found');
-                done();
-            });
+        return request(url, options).then(() => {
+            throw new Error('Request should have errored');
```
Comment (gargol) [inline core/test/unit/lib/request_spec.js]: :+1: this is a lot more readable and cuts unneeded wtfs when debugging :smiley:

Comment (gargol) [reply core/test/unit/lib/request_spec.js]: Was thinking about it a bit more. Would there be a benefit keeping an error messages for these cases more generic/conventional e.g : `should have errored`? This gives enough context on where it failed and from `.catch` we can easily figure out what exactly was supposed to happen? The upside of making the name generic would be ease of grepping for tests that check error handling :thinking:

**Comment by gargol** [comment]
LGTM. @allouis feel free to merge these! The comment that was left is more of an open-ended discussion :)

File: core/test/unit/lib/request_spec.js (hunk 1 above)
Comment (allouis) [reply core/test/unit/lib/request_spec.js]: I had a think about this and i'm on the fence - I'm wondering if it's more useful to be able to search for the ussage of this pattern OR to search for a fairly unique error message... :thinking:

**Pull request merged**